dashboard/client/api/kube-api.ts

```diff
@@ -42,23 +43,65 @@ export interface IKubeApiLinkBase extends IKubeApiLinkRef {
 export class KubeApi<T extends KubeObject = any> {
   static parseApi(apiPath = ""): IKubeApiLinkBase {
     apiPath = new URL(apiPath, location.origin).pathname;
+
     const [, prefix, ...parts] = apiPath.split("/");
     const apiPrefix = `/${prefix}`;
 
-    const [left, right, found] = split(parts, "namespaces");
+    const [left, right, namespaced] = split(parts, "namespaces");
     let apiGroup, apiVersion, namespace, resource, name;
 
-    if (found) {
-      if (left.length == 0) {
-        throw new Error(`invalid apiPath: ${apiPath}`)
+    if (namespaced) {
+      switch (right.length) {
+      case 0:
+        resource = "namespaces"; // special case this due to `split` removing namespaces
+        break;
+      case 1:
+        resource = right[0];
+        break;
+      default:
+        [namespace, resource, name] = right;
+        break;
       }
 
       apiVersion = left.pop();
       apiGroup = left.join("/");
-      [namespace, resource, name] = right;
     } else {
-      [apiGroup, apiVersion, resource] = left;
+      switch (left.length) {
+      case 2:
+        resource = left.pop();
```

Comment:
Forgot `break`?

Reply:
I meant to leave that out, but I will add some more tests once I find which is wrong...